test/test_multitensor.py

```diff
@@ -307,6 +307,24 @@ def test_bn_ast_on_devices(self):
     # for i, ast in enumerate(asts):
     #   print(f"{i} {ast}")
 
+  def test_mlb_assign_change_axis(self):
+    devices = (d0, d1, d2, d3)
+
+    # axis=None <- axis=None + axis=0
+    t_none = Tensor.zeros((16, 16)).shard(devices).realize()  # allocate 16*16 length buffers on each device
+    assert t_none.lazydata.lbs[0].realized.size == 16 * 16  # check that buffers have been allocated
```

Comment:
const multi Tensor will not have realized (similar to single LB case) so this might change